nomad/resource_job.go

```diff
@@ -567,6 +570,32 @@ func parseJobspec(raw string, is_json bool, vaultToken *string) (*api.Job, error
 	return job, nil
 }
 
+func parseJSONJobspec(raw string) (*api.Job, error) {
+	// `nomad job run -output` returns a jobspec with a "Job" root, so
+	// partially parse the input JSON to detect if we have this root.
+	var root map[string]json.RawMessage
+
+	err := json.Unmarshal([]byte(raw), &root)
+	if err != nil {
+		return nil, err
+	}
+
+	jobBytes, ok := root["Job"]
+	if !ok {
+		// Parse the input as is if there's no "Job" root.
+		jobBytes = []byte(raw)
+	}
+
+	// Parse actual job.
+	var job *api.Job
+	err = json.Unmarshal(jobBytes, &job)
```

Comment:
is it necessary to pass a pointer-to-pointer to `Unmarshall` if we don't care about the null case? or necessary at all? this call is so weird.

Reply:
I copied this as it was, but now that you mentioned it does look a bit weird. The pointer-to-pointer is necessary here because otherwise we would be passing `nil` to `json.Unmarshal`.

I think a better version would be:

```go
job := &api.Job{}
err = json.Unmarshal(jobBytes, &job)
```